src/librbd/crypto/CryptoObjectDispatch.cc

```diff
@@ -29,57 +29,66 @@ struct C_EncryptedObjectReadRequest : public Context {
     I* image_ctx;
     CryptoInterface* crypto;
     uint64_t object_no;
-    uint64_t object_off;
-    ceph::bufferlist* read_data;
+    io::ReadExtents* extents;
     Context* onfinish;
     io::ReadResult::C_ObjectReadRequest* req_comp;
     io::ObjectDispatchSpec* req;
 
     C_EncryptedObjectReadRequest(
             I* image_ctx, CryptoInterface* crypto, uint64_t object_no,
-            uint64_t object_off, uint64_t object_len, IOContext io_context,
+            io::ReadExtents* extents, IOContext io_context,
             int op_flags, const ZTracer::Trace &parent_trace,
-            ceph::bufferlist* read_data, int* object_dispatch_flags,
+            uint64_t* version, int* object_dispatch_flags,
             Context** on_finish,
             Context* on_dispatched) : image_ctx(image_ctx),
                                       crypto(crypto),
                                       object_no(object_no),
-                                      object_off(object_off),
-                                      read_data(read_data),
+                                      extents(extents),
                                       onfinish(on_dispatched) {
       *on_finish = util::create_context_callback<
               Context, &Context::complete>(*on_finish, crypto);
 
-      auto aio_comp = io::AioCompletion::create_and_start(
-              (Context*)this, util::get_image_ctx(image_ctx),
-              io::AIO_TYPE_READ);
-      aio_comp->read_result = io::ReadResult{read_data};
-      aio_comp->set_request_count(1);
-
-      auto req_comp = new io::ReadResult::C_ObjectReadRequest(
-              aio_comp, object_off, object_len, {{0, object_len}});
-
       req = io::ObjectDispatchSpec::create_read(
               image_ctx, io::OBJECT_DISPATCH_LAYER_CRYPTO, object_no,
-              {{object_off, object_len}}, io_context, op_flags, parent_trace,
-              &req_comp->bl, &req_comp->extent_map, nullptr, req_comp);
+              &req_comp->extents, io_context, op_flags, parent_trace,
```

Comment:
Nit: I'm probably overlooking something, but where is `req_comp` getting initialized now?

Reply:
You're right, req_comp no longer used. fixing...